### PR TITLE
coverbrowser: make showing some metadata fields optional

### PR DESCRIPTION
**Pull request by benoit-pierre**
Showing the page / progress info and the file size / type can now be disabled.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10941)

### CODE REVIEW

File: plugins/coverbrowser.koplugin/listmenu.lua [hunk 1]
```diff
@@ -423,35 +423,39 @@ function ListMenuItem:update()
...
+                    face = Font:getFace("cfont", fontsize_info),
+                    fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
                 }
-            }
+                table.insert(wright_items, wfileinfo)
+            end
+
+            if not BookInfoManager:getSetting("hide_page_info") then
+                local wpageinfo = TextWidget:new{
+                    text = pages_str or "",
```
Comment (poire-z) [inline plugins/coverbrowser.koplugin/listmenu.lua:442]: Can it happen that pages_str becomes nil ? We initialize it to "" on line 369.

File: plugins/coverbrowser.koplugin/listmenu.lua [hunk 2]
```diff
@@ -423,35 +423,39 @@ function ListMenuItem:update()
...
-            -- the combo centered. Given the nature of our strings (numbers,
-            -- uppercase MB/KB on top text, number and lowercase "page" on bottom text),
-            -- we get the annoying feeling it's not centered but shifted towards top.
-            -- Let's add a small VerticalSpan at top to give a better visual
-            -- feeling of centering.
-            local wright = CenterContainer:new{
-                dimen = Geom:new{ w = wright_width, h = dimen.h },
-                VerticalGroup:new{
-                    align = "right",
-                    VerticalSpan:new{ width = Screen:scaleBySize(2) },
```
Comment (poire-z) [inline plugins/coverbrowser.koplugin/listmenu.lua:450]: You removed my VerticalSpan (without a farewell comment :/). You really didn't get my "annoying feeling it's not centered but shifted towards top" ?
(I just tested with and without on my Kobo, and I somehow don't notice the difference... Dunno if other things had changed - or if I just didn't stare at them enough :)

File: plugins/coverbrowser.koplugin/listmenu.lua (hunk 1 above)
Comment (benoit-pierre) [reply plugins/coverbrowser.koplugin/listmenu.lua]: Amended.

File: plugins/coverbrowser.koplugin/listmenu.lua (hunk 2 above)
Comment (benoit-pierre) [reply plugins/coverbrowser.koplugin/listmenu.lua:450]: There's a 2 pixels difference when simulating a Kindle Paperwhite (before on the left):
results
Measurements:
| | Before | After |
|-|--------|-------|
|1| 36     | 34    |
|2| 31     | 29    |
|3|  4.5   |  6.5  |
|4| 10.5   | 12.5  |
|5| 14.5   | 12.5  |
|6| 19.5   | 17.5  |
|7| 27     | 29    |
|8| 21     | 23    |

IMHO, after is subjectively and objectively better (including simpler code), so I removed the hack.

**Review by poire-z — APPROVED** [review]

**Comment by poire-z** [comment]
@Frenzie : I let you have a look at the new strings - dunno about our usage of `/` between words (size / type, length / progress).

File: plugins/coverbrowser.koplugin/main.lua [hunk 3]
```diff
@@ -192,42 +190,70 @@ function CoverBrowser:addToMainMenu(menu_items)
                     UIManager:show(items)
                 end,
             },
+            {
+                text = _("Length / progress"),
+                sub_item_table = {
+                    {
+                        text = _("Show page / progress info"),
```
Comment (Frenzie) [inline plugins/coverbrowser.koplugin/main.lua:197]: What does this mean? Can't it just be shortened to "show progress"?

File: plugins/coverbrowser.koplugin/main.lua [hunk 4]
```diff
@@ -192,42 +190,70 @@ function CoverBrowser:addToMainMenu(menu_items)
                     UIManager:show(items)
                 end,
             },
+            {
+                text = _("Length / progress"),
```
Comment (Frenzie) [inline plugins/coverbrowser.koplugin/main.lua:194]: We generally prefer to phrase as `Length and progress` if I'm reading this correctly.

File: plugins/coverbrowser.koplugin/main.lua [hunk 5]
```diff
@@ -279,41 +305,14 @@ function CoverBrowser:addToMainMenu(menu_items)
...
-                text = _("Show number of pages read instead of progress %"),
-                checked_func = function() return BookInfoManager:getSetting("show_pages_read_as_progress") end,
-                callback = function()
-                    if BookInfoManager:getSetting("show_pages_read_as_progress") then
-                        BookInfoManager:saveSetting("show_pages_read_as_progress", false)
-                    else
-                        BookInfoManager:saveSetting("show_pages_read_as_progress", true)
-                    end
-                    self:refreshFileManagerInstance()
+                text = _("Show file size / type"),
```
Comment (Frenzie) [inline plugins/coverbrowser.koplugin/main.lua:308]: Show file properties?

**Review by poire-z — APPROVED** [review]

**Pull request merged**